onnxruntime/core/providers/webnn/builders/impl/qdq_op_builder.cc

```diff
@@ -74,31 +77,37 @@ Status QDQOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
                                                              scale,
                                                              emscripten::val::array(target_shape),
                                                              reshape_scale_options);
-    emscripten::val reshape_zero_point_options = emscripten::val::object();
-    reshape_zero_point_options.set("label", node.Name() + "_reshape_zero_point");
-    zero_point = model_builder.GetBuilder().call<emscripten::val>("reshape",
-                                                                  zero_point,
-                                                                  emscripten::val::array(target_shape),
-                                                                  reshape_zero_point_options);
+    if (has_zero_point) {
+      // Insert ones before and after the axis dimension for broadcasting of 1D zero_point tensor.
+      // If zero_point is not provided, leave WebNN to handle the broadcast.
+      emscripten::val reshape_zero_point_options = emscripten::val::object();
+      reshape_zero_point_options.set("label", node.Name() + "_reshape_zero_point");
+      zero_point = model_builder.GetBuilder().call<emscripten::val>("reshape",
+                                                                    zero_point,
+                                                                    emscripten::val::array(target_shape),
+                                                                    reshape_zero_point_options);
+    }
   }
 
-  // If block_size is specified, we need to expand the scale and zero_point tensors.
-  if (block_size > 1) {
-    emscripten::val concat_scale_inputs = emscripten::val::array();
-    emscripten::val concat_zero_point_inputs = emscripten::val::array();
-    for (int i = 0; i < block_size; i++) {
-      concat_scale_inputs.call<void>("push", scale);
-      concat_zero_point_inputs.call<void>("push", zero_point);
+  // If block_size is specified, use WebNN's tile to expand the non-scalar scale and zero_point tensors.
+  if (block_size > 1 && !scale_shape.empty()) {
+    // Now the scale's size is equal to the input's size.
+    std::vector<uint32_t> repeats(input_shape.size(), 1);
+    repeats[axis] = block_size;
```

Comment:
Hmm, I'm not seeing how this works without a reshape. If you have an input of size [4] with values of {1,2,3,4}, a scale of size [2] with values {5,6} (where each element is repeated twice), and repeats of [2], but that would yield an expanded scale of {5,6,5,6} rather than {5,5,6,6}.

I think you'd need a `reshape(tile(reshape(..., scaleAs2D), repeats), originalShape)`, tiling along the lowest dimensions.